craft_application/errors.py

```diff
@@ -140,6 +141,18 @@ def __init__(self) -> None:
         super().__init__(message=message, resolution=resolution)
 
 
+class IncompatibleBaseError(CraftError):
+    """The build plan's base is incompatible with the host environment."""
+
+    def __init__(self, host_base: bases.BaseName, build_base: bases.BaseName) -> None:
```

Comment:
updated, please have a look

Reply:
Looks good, thanks! I'm also okay with having `details` default to `None` unless the caller provides one since it's often more specific to the situation than the error is, but I think you came up with a very good description.